samza/sso-account-updater/src/main/java/org/sunbird/jobs/samza/service/SSOAccountUpdaterService.java

```diff
@@ -160,42 +162,42 @@ private void validateName(Map<String, Object> eventMap, Map<String, Object> upda
     }
 
     private void validateSchool(Map<String, Object> eventMap, Map<String, Object> updateMap, Map<String, Object> passedOrg) {
-        List<String> passedRoles = (List<String>) eventMap.get(SSOAccountUpdaterParams.roles.name());
         String userId = (String) eventMap.get(SSOAccountUpdaterParams.userId.name());
         List<Map<String, Object>> organisations = (List<Map<String, Object>>) eventMap.get(SSOAccountUpdaterParams.organisations.name());
 
-        //if roles is not passed, assume PUBLIC
-        if (CollectionUtils.isEmpty(passedRoles)) {
-            passedRoles = new ArrayList<String>();
-            passedRoles.add(SSOAccountUpdaterParams.PUBLIC.name());
-        }
-
         String passedOrgId = (String) passedOrg.get(SSOAccountUpdaterParams.identifier.name());
         if (CollectionUtils.isEmpty(organisations)) {
-            updateMap.put(SSOAccountUpdaterParams.organisations.name(), getOrganisationsListFromPayloadData(passedOrgId, userId, passedRoles));
+            updateMap.put(SSOAccountUpdaterParams.organisations.name(), getOrganisationsListFromPayloadData(passedOrgId, userId, getPublicRole()));
         } else {
             boolean isUserUpdateRequired = true;
-
-            //ignore PUBLIC role for comparison
-            passedRoles.remove(SSOAccountUpdaterParams.PUBLIC.name());
+            Set<String> existingRoles = new HashSet<String>();
 
             for (Map<String, Object> organisation : organisations) {
+                List<String> assignedRoles = (List<String>) organisation.get(SSOAccountUpdaterParams.roles.name());
```

Comment:
If the given roles list doesn't have PUBLIC, shouldn't we add it?

Reply:
The update/add member API takes care of adding PUBLIC